Formula/l/llvm.rb

```diff
@@ -445,8 +493,38 @@ def install
     end
   end
 
+  # We use the extra layer of indirection in `arch` because the FormulaAudit/OnSystemConditionals
+  # doesn't want to let us use `Hardware::CPU.arch` outside of `install` or `post_install` blocks.
+  def write_config_files(macos_version, kernel_version, arch)
+    clang_config_file_dir.mkpath
+
+    drivers = ["clang", "clang++", "clang-cpp"]
+    arches = Set.new([:arm64, :x86_64])
+    arches << arch
+
+    arches.each do |target_arch|
+      target_triple = "#{target_arch}-apple-darwin#{kernel_version}"
+      drivers.each do |driver|
+        (clang_config_file_dir/"#{target_triple}-#{driver}.cfg").atomic_write <<~CONFIG
+          --sysroot=#{MacOS::CLT::PKG_PATH}/SDKs/MacOSX#{macos_version}.sdk
+        CONFIG
+      end
+    end
+  end
+
+  def post_install
+    return unless OS.mac?
+    # TODO: should we check for all config files here?
+    return if (clang_config_file_dir/"#{Hardware::CPU.arch}-apple-darwin#{OS.kernel_version.major}-clang.cfg").exist?
+
+    write_config_files(MacOS.version.major, OS.kernel_version.major, Hardware::CPU.arch)
+  end
```

Comment:
Was initially wondering if this was really needed given we will pick up new macOS versions 3+ months prior to release but I guess makes sense for the versioned formulae that won't be version bumped.

Feels hacky to make people do `brew postinstall` though. I wonder if it makes sense to either:

* Ship this in `brew` so that `brew update` will be all you need.
* Ship this in a special `clang-config` formula or something that every LLVM formula uses.

Reply:
Well the idea of `postinstall` was for users who didn't get a default config file for whatever reason. The idea is that it's meant to be a rare occurrence, so doing a lot more about it (e.g. files in `brew`, a separate formula) seems overkill.

Reply:
Isn't the point to avoid people needing to avoid `brew reinstall`? It would indeed probably be a rare occurrence for main formula (though rare enough to probably be covered by the unversioned case), but not for versioned formulae which are rarely revision bumped.

Reply:
A versioned formula carrying these changes is still several months away -- I don't think that's worth considering right now. We can handle those properly when it comes to it.

The bigger issue is Xcode-only installs, which will get broken as soon as this is merged: https://github.com/Homebrew/homebrew-core/pull/196094#discussion_r1824091495

Reply:
Will fix this before LLVM 20, but would love more thoughts on the other thread about Xcode-only installs.